libs/client-shared/src/lib/features/alert/alert-list/alert-list.component.ts

```diff
@@ -0,0 +1,18 @@
+import { Component, inject } from '@angular/core';
+import { Store } from '@ngrx/store';
+
+import { AlertId } from '../alert.model';
+import { AlertEntry, AlertState } from '../alert.reducer';
+import { selectAlerts } from '../alert.selectors';
+
+@Component({
+  selector: 'ul[app-alert-list]',
```

Comment:
Same question as above: I have never seen this. What is the benefit over simple selectors?

Reply:
When using an element selector (e.g. `app-alert-list`), we essentially create a new, custom HTML element. This makes sense when there is no existing element that matches the semantic meaning of what we're trying to do. However, if there is such an element, using an attribute selector like here allows us to reuse that element. 

This is especially useful in cases like `li`, which should always follow directly after their parent `ul`/`ol`. With attribute selectors, this is possible, e.g. this is valid HTML:
```html
<ul app-alert-list>
  <li app-alert-item>My Alert</li>
</ul>
```
While this is not:
```html
<app-alert-list>
  <ul>
    <app-alert-item>
      <li>My Alert</li>
    </app-alert-item>
  </ul>
</app-alert-list>
```
Finally, attribute selectors also allow us to generate less HTML when we want to wrap a component's content inside a specific HTML element, just like here. Other elements besides `li` where this is very useful is with things like `button`, `dialog`, `footer`, and more.